lib/avro_turf.rb

```diff
@@ -69,6 +71,31 @@ def decode_stream(stream, schema_name: nil, namespace: @namespace)
     schema = schema_name && @schema_store.find(schema_name, namespace)
     reader = Avro::IO::DatumReader.new(nil, schema)
     dr = Avro::DataFile::Reader.new(stream, reader)
+
+    # If no reader schema is specified, we'll try to find one matching the name
+    # of the writer's schema.
+    if schema_name.nil?
+      schema = load_readers_schema(reader.writers_schema)
+      reader.readers_schema = schema if schema
+    end
+
     dr.first
   end
+
+  private
+
+  # Returns the reader schema matching the writer's schema if one is locally
+  # available, or nil if there's no matching reader schema.
+  def load_readers_schema(writers_schema)
+    return nil unless writers_schema.respond_to?(:fullname)
+
+    schema_name = writers_schema.fullname
+
+    begin
+      @schema_store.find(schema_name)
+    rescue SchemaNotFoundError
+      @logger.warn "Could not find schema `#{schema_name}' locally; using writer's schema instead"
```

Comment:
This might be a decent strategy for the short term, but in the long term, I don't think these errors should be swallowed.  Readers must specify the schema they expect, or they cannot rely on the results.  I don't think readers should be able to opt-out of specifying a schema that they want to use.